account_banking_sepa_direct_debit/models/account_payment_line.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2019 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import UserError
+
+
+class AccountPaymentLine(models.Model):
+    _inherit = 'account.payment.line'
+
+    @api.multi
+    def draft2open_payment_line_check(self):
+        res = super(AccountPaymentLine, self).draft2open_payment_line_check()
+        sepa_dd_lines = self.filtered(
+            lambda l: l.order_id.payment_method_id.code == 'sepa_direct_debit')
+        sepa_dd_lines._check_sepa_direct_debit_ready()
+        return res
+
+    @api.multi
+    def _check_sepa_direct_debit_ready(self):
+        """
+        This method checks whether the payment line(s) are ready to be used
+        in the SEPA Direct Debit file generation.
+        :raise: UserError if a line does not fulfils all requirements
+        """
+        for rec in self:
+            if not rec.mandate_id:
+                raise UserError(_(
+                    "Missing SEPA Direct Debit mandate on the line with "
+                    "partner {partner_name} (reference {reference}).").format(
+                    partner_name=rec.partner_id.name, reference=rec.name))
```

Comment:
BTW, interestingly, this check on mandate_id is already present in account_banking_mandate.

Reply:
I think it's for having a custom message.